src/node/factory.ts

```diff
@@ -47,7 +49,12 @@ export class TurboFactory extends TurboBaseFactory {
         }
       }
     } else if (privateKey) {
-      signer = new ArweaveSigner(privateKey);
+      if (token === 'solana') {
+        // TODO: test wallet creation
+        signer = new HexSolanaSigner(privateKey);
+      } else {
+        signer = new ArweaveSigner(privateKey);
+      }
```

Comment:
from huddle:

- move the below to `/common/factory`
- implement separate function for getting signer

Reply:
[10d0219](https://github.com/ardriveapp/turbo-sdk/pull/113/commits/10d0219dd1d3a038a04e8669c0564e7b3f4b7967)